repo2docker/contentproviders/git.py

```diff
@@ -45,6 +45,12 @@ def fetch(self, spec, output_dir, yield_output=False):
                                     capture=yield_output):
                 yield line
 
+        # ensure that git submodules are initialised and updated
+        for line in execute_cmd(['git', 'submodule', 'update', '--init', '--recursive'],
```

Comment:
Instead of adding an extra step here, does it make sense to use `git clone --recursive` as our clone command? I think that's ~always the right thing to do.